lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,483 @@
+        cube_b = self.fg_cube.copy()
+        cube_c = self.ltng_cube.copy()
+        cube_d = self.precip_cube.copy()
+        cube_e = self.vii_cube.copy()
+        plugin._modify_first_guess(cube_a, cube_b, cube_c, cube_d, cube_e)
+        self.assertArrayAlmostEqual(cube_a.data, self.cube.data)
+        self.assertArrayAlmostEqual(cube_b.data, self.fg_cube.data)
+        self.assertArrayAlmostEqual(cube_c.data, self.ltng_cube.data)
+        self.assertArrayAlmostEqual(cube_d.data, self.precip_cube.data)
+        self.assertArrayAlmostEqual(cube_e.data, self.vii_cube.data)
+
+    def test_precip_zero(self):
+        """Test that ApplyPrecip is being called"""
+        # Set lightning data to zero so it has a Null impact
+        self.ltng_cube.data = np.full_like(self.ltng_cube.data, -1.)
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = set_up_cube_with_no_realizations()
+        expected.data[0, 7, 7] = 0.0067
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_vii_large(self):
+        """Test that ApplyIce is being called"""
+        # Set lightning data to zero so it has a Null impact
+        self.vii_cube.data[:, 7, 7] = 1.
+        self.ltng_cube.data[0, 7, 7] = -1.
+        self.fg_cube.data[0, 7, 7] = 0.
+        # No halo - we're only testing this method.
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        expected.data[0, 7, 7] = 0.9
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            self.vii_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_null(self):
+        """Test that large precip probs and -1 lrates have no impact"""
+        # Set precip data to 0.1, at the top of the upper low range.
+        self.precip_cube.data[0, 0, 7, 7] = 0.1
+        # Set lightning data to -1 so it has a Null impact
+        self.ltng_cube.data = self.ltng_cube.data * 0. - 1.
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = set_up_cube_with_no_realizations(zero_point_indices=[])
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
```

Comment:
If we're testing for "no impact" it would be clearer compare directly with `self.fg_cube.data` rather than duplicating this as a new cube.

Reply:
Redefined "expected" to be a copy of fg_cube.